blockprod/src/detail/mod.rs

```diff
@@ -328,13 +321,33 @@ impl BlockProduction {
         let (job_remover_func, end_confirm_receiver) = self.job_manager.make_job_stopper_function();
         let _job_remover = OnceDestructor::new(move || job_remover_func(job_key));
 
+        let mut previous_attempt = timestamp_at_start;
```

Comment:
"last_timestamp_used"

Also please explain above this the idea behind it. Something in the lines of "Proof of stake depends on timestamp when searching for a valid block. The timestamp starts from the best/previous block's timestamp + 1 second, and ends at the current time stamp + some distance in time defined by the blockchain. This variable keeps track of the latest time that was checked, so that it's not tested multiple times."

Reply:
Yep, done.